python/cudf/cudf/tests/test_array_ufunc.py

```diff
@@ -52,9 +52,18 @@ def _hide_ufunc_warnings(ufunc):
 
 
 @pytest.mark.parametrize("ufunc", _UFUNCS)
-def test_ufunc_index(ufunc):
+def test_ufunc_index(request, ufunc):
     # Note: This test assumes that all ufuncs are unary or binary.
     fname = ufunc.__name__
+    request.applymarker(
+        pytest.mark.xfail(
+            condition=(
+                fname in {"bitwise_and", "bitwise_or", "bitwise_xor"}
+                and not PANDAS_GE_150
```

Comment:
Historically we have been doing this version-based xfailing when we do an upgrade in cudf, but the question here seems to be how do we phase out the very very old versions we don't test or match against anymore. I'm in favor of stripping away very old version-based xfails, but we should probably discuss the pros and cons of doing so with internal teams too.
My pros:
1. Quality of life improvement and lesser ambiguity in tests.
2. Easy to review

Cons:
1. Someone who tests latest cudf with an old version of pandas(say 1.0) will not be able to get all xfails passing. But the tests that are not being xfailed too will anyway fail because our latest cudf is actually trying to match pandas-1.5 rather than pandas-1.0, so this isn't a huge issue. 

Added this to our weekly agenda to discuss what the team feels about on this issue. @quasiben ran into this recently too.

Reply:
Are our tests considered user facing? IIRC this was invented to get us through some kind of complex CI situation where different test containers ended up pulling different versions of pandas that were allowed within the window. To me, support for a certain range of pandas versions means "your cuDF code will work and produce the same answer, no matter what version of pandas you use within this window" and not much about our unit testing. 

I think viewed from that lens, what this is pointing to is that even though it's very convenient to use pandas to generate "expected" values in most of our tests, it's not actually right to do that because we're aiming for a moving target. Until we stop doing that I suspect we might end up needing flags like this from time to time. I'd rather see us long term take those tests and really take some time to define together with pandas exactly what that API should do, and then write pandas out of the test completely.

Reply:
Agreed. #6435 is relevant here too.